aicsimageio/aics_image.py

```diff
@@ -20,14 +21,45 @@ class AICSImage:
     # Example:
     # if TiffReader was placed before OmeTiffReader,
     # we would never hit the OmeTiffReader
-    # SUPPORTED_READERS = [
-    #     ArrayLikeReader,
-    #     CziReader,
-    #     LifReader,
-    #     OmeTiffReader,
-    #     TiffReader,
-    #     DefaultReader,
-    # ]
+    SUPPORTED_READERS = (
+        # readers.ArrayLikeReader,
+        # readers.CziReader,
+        # readers.LifReader,
+        readers.OmeTiffReader,
+        readers.TiffReader,
+        readers.DefaultReader,
+    )
+
+    @staticmethod
+    def determine_reader(image: types.ImageLike, **kwargs) -> Reader:
+        """
+        Cheaply check to see if a given file is a recognized type and return the
+        appropriate reader for the image.
+
+        Returns
+        -------
+        ReaderClass: Reader
+            The reader that supports the provided image.
+        kwargs: Any
+            Any kwargs used for reading and validation of the file.
+
+        Raises
+        ------
+        exceptions.UnsupportedFileFormatError
+            No reader could be found that supports the provided image.
+        """
+        for ReaderClass in AICSImage.SUPPORTED_READERS:
+            if ReaderClass.is_supported_image(image, **kwargs):
+                return ReaderClass
+
+        # Construct non-URI image "paths"
+        # numpy, dask, etc.
+        if isinstance(image, (str, Path)):
+            path = str(image)
+        else:
+            path = type(image)
+
+        raise exceptions.UnsupportedFileFormatError("AICSImage", path)
```

Comment:
This will result in things like `UnsupportedFileFormatError: "AICSImage object does not support: 'dict'"`

Will add tests for this when I add the `ArrayLikeReader`.